crml/transaction-payment/src/lib.rs

```diff
@@ -102,12 +113,16 @@ impl<T: Trait> Module<T> {}
 /// Require the transactor pay for themselves and maybe include a tip to gain additional priority
 /// in the queue.
 #[derive(Encode, Decode, Clone, Eq, PartialEq)]
-pub struct ChargeTransactionPayment<T: Trait + Send + Sync>(#[codec(compact)] BalanceOf<T>);
+pub struct ChargeTransactionPayment<T: Trait + Send + Sync> {
+	#[codec(compact)]
+	tip: BalanceOf<T>,
+	fee_exchange: Option<FeeExchange<BalanceOf<T>>>,
```

Comment:
This should be the versioned enum not `FeeExchangeV1` directly, that can be matched over when it's used in `fn validate`